polkadot/xcm/src/v4/mod.rs

```diff
@@ -1280,6 +1280,9 @@ impl<Call> TryFrom<NewXcm<Call>> for Xcm<Call> {
 	}
 }
 
+// We use a big weight here but not `Weight::MAX` as a best effort.
+const DEFAULT_WEIGHT_FOR_TRANSACT_CONVERSION: Weight = Weight::from_parts(10_000_000_000, 1_000_000);
```

Comment:
I'm not sure, but do we have an indexer where we can check/monitor all `Transact` instructions in live chains? It would be interesting to compare these values with `Weight::from_parts(10_000_000_000, 1_000_000).`

Just out of curiosity, how was this `10_000_000_000` value determined? At least, some docs for that would be cool. Does it truly cover all xcmV4 scenarios for everyone? Maybe we could set this value based on some block/extrinsic size heuristics, like: _"we know that max block/extrinsic size is approximately XYZ on relays/parachains, so let's set this as a e.g. 75% of some actual max value."_ or something like that.

https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fkusama-rpc.dwellir.com#/chainstate/constants
`system.blockWeights`:
```
{
  baseBlock: {
    refTime: 14,278,073,000
    proofSize: 0
  }
  maxBlock: {
    refTime: 2,000,000,000,000
    proofSize: 18,446,744,073,709,551,615
  }
  perClass: {
    normal: {
      baseExtrinsic: {
        refTime: 124,706,000
        proofSize: 0
      }
      maxExtrinsic: {
        refTime: 1,479,875,294,000
        proofSize: 13,650,590,614,545,068,195
      }
      maxTotal: {
        refTime: 1,500,000,000,000
        proofSize: 13,835,058,055,282,163,711
      }
      reserved: {
        refTime: 0
        proofSize: 0
      }
    }
    operational: {
      baseExtrinsic: {
        refTime: 124,706,000
        proofSize: 0
      }
      maxExtrinsic: {
        refTime: 1,979,875,294,000
        proofSize: 18,262,276,632,972,456,099
      }
      maxTotal: {
        refTime: 2,000,000,000,000
        proofSize: 18,446,744,073,709,551,615
      }
      reserved: {
        refTime: 500,000,000,000
        proofSize: 4,611,686,018,427,387,904
      }
    }
    mandatory: {
      baseExtrinsic: {
        refTime: 124,706,000
        proofSize: 0
      }
      maxExtrinsic: null
      maxTotal: null
      reserved: null
    }
  }
}
```

I don't know, but do we have some indexer where we could see all `Transact` instructions live? At least, it would be cool to compare those values with `Weight::from_parts(10_000_000_000, 1_000_000)`.

Reply:
It's a placeholder value that wasn't meant to go through. I'll do the perfect conversion with `get_dispatch_info().call_weight` so the conversion never fails.

Reply:
yes, let's do the conversion using the exact required weight from `get_dispatch_info().call_weight`